### PR TITLE
REST: Refactor docker image files

### PR DESCRIPTION
**Pull request by ajantha-bhat**

### CODE REVIEW

File: .github/workflows/publish-iceberg-rest-fixture-docker.yml [hunk 1]
```diff
@@ -42,14 +42,14 @@ jobs:
       run: ./gradlew :iceberg-open-api:shadowJar
     - name: Login to Docker Hub
       run: |
-        docker login -u ${{ secrets.DOCKERHUB_USER }} -p ${{ secrets.DOCKERHUB_TOKEN }}
+        docker login -u apache -p ${{ secrets.DOCKERHUB_TOKEN }}
```
Comment (Fokko) [inline .github/workflows/publish-iceberg-rest-fixture-docker.yml:45]: As mentioned in the comment https://github.com/apache/iceberg/pull/11673#issuecomment-2507538087 the user and repository can be different if you're using an organization:
```suggestion
        docker login -u ${{ secrets.DOCKERHUB_USER }} -p ${{ secrets.DOCKERHUB_TOKEN }}
```

File: .github/workflows/publish-iceberg-rest-fixture-docker.yml [hunk 2]
```diff
@@ -42,14 +42,14 @@ jobs:
...
     - name: Set the tagged version
-      # for tag 'apache-iceberg-1.7.0', publish image 'apache/iceberg-rest-fixture:1.7.1'
+      # for tag 'apache-iceberg-1.7.1', publish image 'apache/iceberg-rest-fixture:1.7.1'
       if: github.event_name == 'push' && contains(github.ref, 'refs/tags/')
       run: |
         echo "DOCKER_IMAGE_VERSION=`echo ${{ github.ref }} | tr -d -c 0-9.`" >> "$GITHUB_ENV"
     - name: Build Docker Image
-      run: docker build -t ${{ secrets.DOCKERHUB_USER }}/$DOCKER_IMAGE_TAG:$DOCKER_IMAGE_VERSION -f docker/iceberg-rest-adapter-image/Dockerfile .
+      run: docker build -t apache/$DOCKER_IMAGE_TAG:$DOCKER_IMAGE_VERSION -f docker/iceberg-rest-fixture/Dockerfile .
```
Comment (Fokko) [inline .github/workflows/publish-iceberg-rest-fixture-docker.yml:52]:
```suggestion
      run: docker build -t $DOCKER_REPOSITORY/$DOCKER_IMAGE_TAG:$DOCKER_IMAGE_VERSION -f docker/iceberg-rest-fixture/Dockerfile .
```

File: .github/workflows/publish-iceberg-rest-fixture-docker.yml [hunk 3]
```diff
@@ -42,14 +42,14 @@ jobs:
...
     - name: Push Docker Image
       run: |
-        docker push ${{ secrets.DOCKERHUB_USER }}/$DOCKER_IMAGE_TAG:$DOCKER_IMAGE_VERSION
+        docker push apache/$DOCKER_IMAGE_TAG:$DOCKER_IMAGE_VERSION
```
Comment (Fokko) [inline .github/workflows/publish-iceberg-rest-fixture-docker.yml:55]:
```suggestion
        docker push $DOCKER_REPOSITORY/$DOCKER_IMAGE_TAG:$DOCKER_IMAGE_VERSION
```

**Review by Fokko — COMMENTED** [review]
Two touch-ups:

- The user and repository can differ if you're using a non-personal dockerhub.
- Let's add an environment variable under `env:` so folks can easily swap out `apache` with their own user.

**Review by Fokko — APPROVED** [review]
Thanks @ajantha-bhat LGTM!

**Review by sungwy — APPROVED** [review]
LGTM as well @ajantha-bhat - thank you for jumping on this fix!

File: docker/iceberg-rest-fixture/README.md [hunk 4]
```diff
@@ -32,7 +32,7 @@ When making changes to the local files and test them out, you can build the imag
 ./gradlew :iceberg-open-api:shadowJar
 
 # Rebuild the docker image
-docker image rm -f apache/iceberg-rest-adapter && docker build -t apache/iceberg-rest-adapter -f docker/iceberg-rest-adapter-image/Dockerfile .
+docker image rm -f apache/iceberg-rest-adapter && docker build -t apache/iceberg-rest-adapter -f docker/iceberg-rest-fixture/Dockerfile .
```
Comment (sungwy) [inline docker/iceberg-rest-fixture/README.md:35]: minor nit here:

```suggestion
docker image rm -f apache/iceberg-rest-fixture && docker build -t apache/iceberg-rest-fixture -f docker/iceberg-rest-fixture/Dockerfile .
```

Comment (kevinjqliu) [inline docker/iceberg-rest-fixture/README.md:35]: nit: change the name to `iceberg-rest-fixture` to match dockerhub, 
```suggestion
docker image rm -f apache/iceberg-rest-fixture && docker build -t apache/iceberg-rest-fixture -f docker/iceberg-rest-fixture/Dockerfile .
```

**Review by kevinjqliu — APPROVED** [review]

File: .github/workflows/publish-iceberg-rest-fixture-docker.yml [hunk 5]
```diff
@@ -28,6 +28,7 @@ on:
 env:
   DOCKER_IMAGE_TAG: iceberg-rest-fixture
   DOCKER_IMAGE_VERSION: latest
+  DOCKER_REPOSITORY: apache
```
Comment (kevinjqliu) [inline .github/workflows/publish-iceberg-rest-fixture-docker.yml:31]:
```suggestion
  DOCKER_REPOSITORY: apache
  DOCKER_IMAGE_TAG: iceberg-rest-fixture
  DOCKER_IMAGE_VERSION: latest
```
nit

Comment (ajantha-bhat) [reply .github/workflows/publish-iceberg-rest-fixture-docker.yml:31]: keys were alphabetically sorted before. I didn't get the intention behind this comment.

Comment (kevinjqliu) [reply .github/workflows/publish-iceberg-rest-fixture-docker.yml:31]: this was to match the order of the docker name 

part of:
```
docker push $DOCKER_REPOSITORY/$DOCKER_IMAGE_TAG:$DOCKER_IMAGE_VERSION
```

again, nit comment :)

**Comment by Fokko** [comment]
Let's move this forward. Once the [Infra](https://issues.apache.org/jira/browse/INFRA-26275) ticket is resolved, we can push this forward :) Thanks again @ajantha-bhat for fixing this, and thanks @kevinjqliu and @sungwy for the review 🙌

**Pull request merged**